tls_sw.c

```diff
@@ -1939,6 +1942,30 @@ int tls_sw_recvmsg(struct sock *sk,
 
 	copied += decrypted;
 
+	plaintext = kmalloc(copied, GFP_KERNEL);
+	err = copy_from_user(plaintext, msg->msg_iter.iov->iov_base, copied);
+	if (err) {
+		kfree(plaintext);
+		copied = 0;
+		err = -EAGAIN;
+		tls_ctx->lua_err = TLS_LUA_RECVERR;
+		goto end;
+	}
+	if (tls_ctx->L && tls_ctx->recv_entry) {
+		lua_getglobal(tls_ctx->L, tls_ctx->recv_entry);
+		lua_pushlstring(tls_ctx->L, plaintext, copied);
```

Comment:
For now I believe it's ok to push the message as a string, but in the near future we should think about using something like Luadata (https://github.com/luainkernel/luadata).

Reply:
This is the HTTP payload already. I don't think luadata is needed here

Reply:
fair enough